src/dbnode/storage/series/buffer.go

```diff
@@ -966,6 +975,22 @@ func (b *BufferBucketVersions) streams(ctx context.Context, opts streamsOptions)
 	return res
 }
 
+func (b *BufferBucketVersions) firstWriteAt(opts streamsOptions) time.Time {
+	var res time.Time
```

Comment:
nit: can shortcircuit on `!opts.filterWriteType` without reading buckets

Reply:
If I don't need to filter the write types than it should read all the buckets to be consistent w/ the current logic. I changed the expression to: 
```
if opts.filterWriteType && bucket.writeType != opts.writeType {
        continue
}
```